### ISSUE TITLE
Running on target (HW)

### ISSUE THREAD

**Issue by asafravid** [ipdk-io/ipdk#24]
When running `make volume` and then going through the steps on a host machine, the process is quite long as many libraries are cloned, processed, built, etc

For a host machine that takes ~20-30 minutes to reach step `5)` in the Readme.

For the target HW, how (if at all) can this be skipped such that all is ready (cloned and **cross-**compiled for arm) for the `Section 2: Example scenario setup ( VM1 <-> IPDK Container <-> VM2 )` tests?

In other words, 
1. Does the HW target need to clone, make, process (etc) all the steps in order to have the p4 ovs ready?
2. Is there a way to prepare the docker image on host and then simply run it (with ovs ready and running) on a HW target DPU? 
3. Otherwise, as I understand, I should have Ubuntu20.04 (per available example in repo) running on the Target HW - correct?
3. Did you guys test this on any HW (such as Marvell / Melanox DPUs)?

**Comment by mestery** [comment]
Correct me if I'm wrong, but I think there are two issues you present in this issue:

1. Decoupling building everything from running everything. E.g. have a step which builds a container which you can launch without having to do the build steps.
2. Run this on target hardware.

For 1 above, on slack I see Sander Tolsma has indicated he plans to push a patch to do this.

For 2, I'll let some other folks comment, but I'd also like to understand how the existing IPDK runs on hardware components as well.

**Comment by asafravid** [comment]
@mestery yes you summarized it well

**Comment by stolsma** [comment]
> For the target HW, how (if at all) can this be skipped such that all is ready (cloned and **cross-**compiled for arm) for the `Section 2: Example scenario setup ( VM1 <-> IPDK Container <-> VM2 )` tests?
> 
> In other words,
> 
>     1. Does the HW target need to clone, make, process (etc) all the steps in order to have the p4 ovs ready?
> 
>     2. Is there a way to prepare the docker image on host and then simply run it (with ovs ready and running) on a HW target DPU?
> 
>     3. Otherwise, as I understand, I should have Ubuntu20.04 (per available example in repo) running on the Target HW - correct?
> 
>     4. Did you guys test this on any HW (such as Marvell / Melanox DPUs)?

@asafravid Currently no HW is supported, they (Intel) are just setting up the skelleton environment and started with P4 supporting ovs version as software switch on x86. I'm expecting that HW supporting IPDK will take some more time before being added here as Intel is still developing them. Also I don't know if Marvel and Mellanox will be supported here,  that will be up to them I pressume....

BTW: can you say something about what you want to achieve by using IPDK? For me it is a performant P4 programmable software vSwitch that can be managed like OVS. Hardware offload support with the same API is the next goal after that.

**Comment by asafravid** [comment]
@stolsma to be more specific, my intention is to run the container with the SW solution (container running `p4 ovs`) of `IPDK` on arm64 cores (`Ubuntu20.04` of course), so was wondering if that was tested/tried prior.

**Comment by stolsma** [comment]
@asafravid  Nope, thats not tested as the P4 ovs version (which is backed by DPDK SWX code) doesn't support Arm cores yet! (because DPDK SWX doesn't support Arm). I'm sure that support will be coming soon because the Intel Mount Evans IPU is equipped Arm N1 cores...

**Comment by namratalimaye** [comment]
1. Does the HW target need to clone, make, process (etc) all the steps in order to have the p4 ovs ready?
Nam) yes

2. Is there a way to prepare the docker image on host and then simply run it (with ovs ready and running) on a HW target DPU?
Nam) You will have to build P4-OVS with the HW target library similar to DPDK target - https://github.com/p4lang/p4-dpdk-target and implement all APIs to run for target

3. Otherwise, as I understand, I should have Ubuntu20.04 (per available example in repo) running on the Target HW - correct?
Nam) Yes, currently only Ubuntu 18, 20 and Fedora is supported.

4. Did you guys test this on any HW (such as Marvell / Melanox DPUs)?
Nam) No, but we have run on Tofino. Tofino uses Ubuntu 20.

Thanks

**Comment by asafravid** [comment]
Closing.
Slowness due to NFS server used as storage
